odoo/addons/base/models/ir_actions.py

```diff
@@ -135,6 +135,34 @@ def get_bindings(self, model_name):
             result['action'] = sorted(result['action'], key=lambda vals: vals.get('sequence', 0))
         return result
 
+    @api.model
+    def _for_xml_id(self, module, xml_id):
+        """ Returns the action content for the provided xml_id
+
+        :param module: the module the act_window originates in
+        :param xml_id: the namespace-less id of the action (the @id
+                       attribute from the XML file)
+        :return: A read() view of the ir.actions.action safe for web use
+        """
+        record = self.env.ref("%s.%s" % (module, xml_id))
+        assert isinstance(self.env[record._name], type(self))
+        action = record.sudo().read()[0]
+        return {
+            field: value
+            for field, value in action.items()
+            if field in record._get_readable_fields()
+        }
+
+    def _get_readable_fields(self):
+        """ return the list of fields that are safe to read
+
+        Fetched via /web/action/load or _for_xml_id method
+        """
+        return set([
+            "binding_model_id", "binding_type", "binding_view_types",
+            "display_name", "help", "id", "name", "type", "xml_id",
+        ])
```

Comment:
Python 3.6 has a notation for sets
```suggestion
        return {
            "binding_model_id", "binding_type", "binding_view_types",
            "display_name", "help", "id", "name", "type", "xml_id",
        }
```